bond_calculator.py

```diff
@@ -115,7 +112,14 @@ def calc_macaulay_duration(self, bond, yld):
         time to cashflow weighted by PV
         """
         # TODO: implement details here - Ching Kung
+        # wavg weighted average term to maturity
+        wavg = ((1 + yld) / (bond.payment_freq * yld) - ((1+ yld + bond.term * (bond.coupon - yld))))
+
+        # PVs = Current Bond Price
+        PVs = ((bond.payment_freq * bond.coupon * ((1 + yld)**bond.term - 1)) + bond.payment_freq * yld)

```

Comment:
PVs is actually calculated by other function, you can simply do `PVs = self.calc_clean_price(bond, yld)`